lib/rex/post/postgresql/ui/console/interactive_channel.rb

```diff
@@ -0,0 +1,160 @@
+# -*- coding: binary -*-
+module Rex
+module Post
+module PostgreSQL
+module Ui
+
+###
+#
+# Mixin that is meant to extend the base channel class from meterpreter in a
+# manner that adds interactive capabilities.
+#
+###
+module Console::InteractiveChannel
+
+  include Rex::Ui::Interactive
+
+  #
+  # Interacts with self.
+  #
+  def _interact
+    begin
+      while self.interacting
+        raw_sql_string = _get_user_input
+        # We need to check that the user is still interacting, i.e. if ctrl+z is triggered when requesting user input
+        break unless (self.interacting && raw_sql_string)
+
+        formatted_query = client_dispatcher.process_query(query: raw_sql_string)
+        print_status "Executing query: #{formatted_query}"
+        client_dispatcher.cmd_query(formatted_query)
+      end
+    rescue ::StandardError => e
+      elog('Error when interact with SQL REPL', e)
+      raise e
+    end
+  end
+
+  #
+  # Called when an interrupt is sent.
+  #
+  def _interrupt
+    prompt_yesno('Terminate interactive REPL interpreter?')
+  end
+
+  #
+  # Suspends interaction with the interactive REPL interpreter
+  #
+  def _suspend
+    if (prompt_yesno('Background interactive REPL interpreter?') == true)
+      self.interacting = false
+    end
+  end
+
+  #
+  # We don't need to do any clean-up when finishing the interaction with the REPL
+  #
+  def _interact_complete
+    # noop
+  end
+
+  # To have this be consistent between reline and the fallback, this will return a single string only.
+  # Try getting multi-line input support provided by Reline, fall back to Readline.
+  def _multiline_with_fallback
+    begin
+      query = _multiline
+      query = _fallback if query[:status] == :fail
+    rescue ::StandardError => e
+      elog('Failed to get SQL input from user', e)
+      raise e
+    end
+
+    query[:result]
+  end
+
+  def _multiline
```

Comment:
Was there a reason for implementing this here, versus over in the `Rex::Ui::Text::Input::Readline` module beside `pgets`? 👀